Sources/MockService.swift

```diff
@@ -264,6 +330,56 @@ private extension MockService {
 		}
 	}
 
+	@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+	func setupPactInteraction(timeout: TimeInterval, file: FileString?, line: UInt?, mockServer: MockServer, testFunction: @escaping (String) async throws -> Void) async throws {
+		Logger.log(message: "Setting up pact test", data: pact.data)
+		do {
+			// Set up a Mock Server with Pact data and on desired http protocol
+			try await mockServer.setup(pact: pact.data!, protocol: transferProtocolScheme)
+			
+			// If Mock Server spun up, run the test function
+			let task = Task(timeout: timeout) {
+				do {
+					try await testFunction(mockServer.baseUrl)
+				} catch {
+					self.failWith("🚨 Error thrown in test function: \(error.localizedDescription)", file: file, line: line)
+					throw error
```

Comment:
Hrm… Will this error just be caught by the `catch` below on 351?

Reply:
Yeah, it will be. I meant for that to do that.  It might not be a good idea though. What I'm trying to do here is have any error thrown in the `testFunction` be reported as such and then also bail out. It is not idea.

Open to suggestions for improvement.